src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2534,7 +2546,104 @@ namespace winrt::TerminalApp::implementation
     {
         for (uint32_t i = 0; i < _tabs.Size(); ++i)
         {
-            _GetStrongTabImpl(i)->UpdateTabViewIndex(i);
+            _tabs.GetAt(i).UpdateTabViewIndex(i);
+        }
+    }
+
+    void TerminalPage::_OpenSettingsUI()
+    {
+        // If we're holding the settings tab's switch command, don't create a new one, switch to the existing one.
+        if (!_switchToSettingsCommand)
+        {
+            // Add the new tab to the list of our tabs.
+            auto newTabImpl = winrt::make_self<SettingsTab>();
+            _tabs.Append(*newTabImpl);
+
+            // Give the tab its index in the _tabs vector so it can manage its own SwitchToTab command.
+            newTabImpl->UpdateTabViewIndex(_tabs.Size() - 1);
+
+            // Don't capture a strong ref to the tab. If the tab is removed as this
+            // is called, we don't really care anymore about handling the event.
+            auto weakTab = make_weak(newTabImpl);
+
+            auto tabViewItem = newTabImpl->TabViewItem();
+            _tabView.TabItems().Append(tabViewItem);
```

Comment:
Is this because we manually keep `_tabs` and `_tabView` in sync? :(

Reply:
exactly 😢